### ISSUE TITLE
Install issue

### ISSUE THREAD

**Issue by cutemonster** [aimeos/aimeos#86]
I get to a point right after the first migration is finished and when running the **App\Composer::setup**
```
 [TypeError]
  Argument 1 passed to Symfony\Component\Process\Process::__construct() must be of the type array, string given, called in C:\WEB\myshop\app\Composer.php on line 210
```
Can anyone tell me how to fix this?

**Comment by aimeos** [comment]
Can you post the output of `composer show`?

**Comment by cutemonster** [comment]
Please see below

```
aimeos/ai-admin-jqadm              2021.10.15 Aimeos Vue.js+Bootstrap admin interface
aimeos/ai-admin-jsonadm            2021.10.4  Aimeos ai-admin-jsonadm extension
aimeos/ai-client-html              2021.10.14 Aimeos ai-client-html extension
aimeos/ai-client-jsonapi           2021.10.4  Aimeos JSON API extension
aimeos/ai-cms-grapesjs             2021.10.7  Aimeos GrapesJS CMS extension
aimeos/ai-controller-frontend      2021.10.4  Aimeos ai-controller-frontend extension
aimeos/ai-controller-jobs          2021.10.8  Aimeos ai-controller-jobs extension
aimeos/ai-gettext                  2021.10.3  Aimeos Gettext extension
aimeos/ai-laravel                  2021.10.3  Laravel adapter for Aimeos web shops and e-commerce solutions
aimeos/ai-swiftmailer              2021.10.2  SwiftMailer adapter for Aimeos web shops and e-commerce solutions
aimeos/aimeos-core                 2021.10.16 Full-featured e-commerce components for high performance online shops
aimeos/aimeos-laravel              2021.10.5  Professional, full-featured and high performance Laravel e-commerce package for online shops and complex B2B projects
aimeos/map                         2.6.1      Easy and elegant handling of PHP arrays as array-like map objects similar to jQuery and Laravel Collections
asm89/stack-cors                   v2.1.1     Cross-origin resource sharing library and stack middleware
brick/math                         0.9.3      Arbitrary-precision arithmetic library
doctrine/cache                     2.1.1      PHP Doctrine Cache library is a popular cache implementation that supports many different drivers such as redis, memcache, apc, mongodb and others.
doctrine/dbal                      2.13.8     Powerful PHP database abstraction layer (DBAL) with many features for database schema introspection and management.
doctrine/deprecations              v0.5.3     A small layer on top of trigger_error(E_USER_DEPRECATED) or PSR-3 logging with options to disable all deprecations or selectively for packages.
doctrine/event-manager             1.1.1      The Doctrine Event Manager is a simple PHP event system that was built to be used with the various Doctrine projects.
doctrine/inflector                 2.0.4      PHP Doctrine Inflector is a small library that can perform string manipulations with regard to upper/lowercase and singular/plural forms of words.
doctrine/instantiator              1.4.1      A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                     1.2.3      PHP Doctrine Lexer parser library that can be used in Top-Down, Recursive Descent Parsers.
dragonmantank/cron-expression      v3.3.1     CRON for PHP: Calculate the next or previous run date and determine if a CRON expression is due
egulias/email-validator            2.1.25     A library for validating emails against several RFCs
enshrined/svg-sanitize             0.15.4     An SVG sanitizer for PHP
facade/flare-client-php            1.9.1      Send PHP errors to Flare
facade/ignition                    2.17.5     A beautiful error page for Laravel applications.
facade/ignition-contracts          1.0.2      Solution contracts for Ignition
fakerphp/faker                     v1.19.0    Faker is a PHP library that generates fake data for you.
filp/whoops                        2.14.5     php error handling for cool kids
fruitcake/laravel-cors             v2.2.0     Adds CORS (Cross-Origin Resource Sharing) headers support in your Laravel application
graham-campbell/result-type        v1.0.4     An Implementation Of The Result Type
guzzlehttp/guzzle                  7.4.2      Guzzle is a PHP HTTP client library
guzzlehttp/promises                1.5.1      Guzzle promises library
guzzlehttp/psr7                    2.2.1      PSR-7 message implementation that also provides common utility methods
hamcrest/hamcrest-php              v2.0.1     This is the PHP port of Hamcrest Matchers
laminas/laminas-diactoros          2.9.0      PSR HTTP Message implementations
laravel/framework                  v8.83.6    The Laravel Framework.
laravel/sail                       v1.13.8    Docker files for running a basic Laravel application.
laravel/sanctum                    v2.15.0    Laravel Sanctum provides a featherweight authentication system for SPAs and simple APIs.
laravel/serializable-closure       v1.1.1     Laravel Serializable Closure provides an easy and secure way to serialize closures in PHP.
laravel/tinker                     v2.7.2     Powerful REPL for the Laravel framework.
laravel/ui                         v3.4.5     Laravel UI utilities and presets.
league/commonmark                  1.6.7      Highly-extensible PHP Markdown parser which fully supports the CommonMark spec and Github-Flavored Markdown (GFM)
league/flysystem                   1.1.9      Filesystem abstraction: Many filesystems, one API.
league/mime-type-detection         1.9.0      Mime-type detection for Flysystem
mockery/mockery                    1.5.0      Mockery is a simple yet flexible PHP mock object framework
monolog/monolog                    2.4.0      Sends your logs to files, sockets, inboxes, databases and various web services
myclabs/deep-copy                  1.11.0     Create deep copies (clones) of your objects
nesbot/carbon                      2.57.0     An API extension for DateTime that supports 281 different languages.
nikic/php-parser                   v4.13.2    A PHP parser written in PHP
nunomaduro/collision               v5.11.0    Cli error handling for console/command-line PHP applications.
nyholm/psr7                        1.5.0      A fast PHP7 implementation of PSR-7
opis/closure                       3.6.3      A library that can be used to serialize closures (anonymous functions) and arbitrary objects.
phar-io/manifest                   2.0.3      Component for reading phar.io manifest information from a PHP Archive (PHAR)
phar-io/version                    3.2.1      Library for handling version information and constraints
php-http/message-factory           v1.0.2     Factory interfaces for PSR-7 HTTP Message
phpdocumentor/reflection-common    2.2.0      Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock  5.3.0      With this component, a library can provide support for annotations via DocBlocks or otherwise retrieve information that is embedded in a DocBlock.
phpdocumentor/type-resolver        1.6.1      A PSR-5 based resolver of Class names, Types and Structural Element Names
phpoption/phpoption                1.8.1      Option Type for PHP
phpspec/prophecy                   v1.15.0    Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage          9.2.15     Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator          3.0.6      FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-invoker                3.1.1      Invoke callables with a timeout
phpunit/php-text-template          2.0.4      Simple template engine.
phpunit/php-timer                  5.0.3      Utility class for timing
phpunit/phpunit                    9.5.19     The PHP Unit Testing framework.
psr/container                      1.1.1      Common Container Interface (PHP FIG PSR-11)
psr/event-dispatcher               1.0.0      Standard interfaces for event handling.
psr/http-client                    1.0.1      Common interface for HTTP clients
psr/http-factory                   1.0.1      Common interfaces for PSR-7 HTTP message factories
psr/http-message                   1.0.1      Common interface for HTTP messages
psr/log                            1.1.4      Common interface for logging libraries
psr/simple-cache                   1.0.1      Common interfaces for simple caching
psy/psysh                          v0.11.2    An interactive shell for modern PHP.
ralouphie/getallheaders            3.0.3      A polyfill for getallheaders.
ramsey/collection                  1.2.2      A PHP library for representing and manipulating collections.
ramsey/uuid                        4.2.3      A PHP library for generating and working with universally unique identifiers (UUIDs).
sebastian/cli-parser               1.0.1      Library for parsing CLI options
sebastian/code-unit                1.0.8      Collection of value objects that represent the PHP code units
sebastian/code-unit-reverse-lookup 2.0.3      Looks up which function or method a line of code belongs to
sebastian/comparator               4.0.6      Provides the functionality to compare PHP values for equality
sebastian/complexity               2.0.2      Library for calculating the complexity of PHP code units
sebastian/diff                     4.0.4      Diff implementation
sebastian/environment              5.1.3      Provides functionality to handle HHVM/PHP environments
sebastian/exporter                 4.0.4      Provides the functionality to export PHP variables for visualization
sebastian/global-state             5.0.5      Snapshotting of global state
sebastian/lines-of-code            1.0.3      Library for counting the lines of code in PHP source code
sebastian/object-enumerator        4.0.4      Traverses array structures and object graphs to enumerate all referenced objects
sebastian/object-reflector         2.0.4      Allows reflection of object attributes, including inherited and non-public ones
sebastian/recursion-context        4.0.4      Provides functionality to recursively process PHP variables
sebastian/resource-operations      3.0.3      Provides a list of PHP built-in functions that operate on resources
sebastian/type                     3.0.0      Collection of value objects that represent the types of the PHP type system
sebastian/version                  3.0.2      Library that helps with managing the version number of Git-hosted PHP projects
swiftmailer/swiftmailer            v6.3.0     Swiftmailer, free feature-rich PHP mailer
symfony/console                    v5.4.5     Eases the creation of beautiful and testable command line interfaces
symfony/css-selector               v5.4.3     Converts CSS selectors to XPath expressions
symfony/deprecation-contracts      v2.5.0     A generic function and convention to trigger deprecation notices
symfony/error-handler              v5.4.3     Provides tools to manage errors and ease debugging PHP code
symfony/event-dispatcher           v5.4.3     Provides tools that allow your application components to communicate with each other by dispatching events and listening to them
symfony/event-dispatcher-contracts v2.5.0     Generic abstractions related to dispatching event
symfony/finder                     v5.4.3     Finds files and directories via an intuitive fluent interface
symfony/http-foundation            v5.4.6     Defines an object-oriented layer for the HTTP specification
symfony/http-kernel                v5.4.6     Provides a structured process for converting a Request into a Response
symfony/mime                       v5.4.3     Allows manipulating MIME messages
symfony/polyfill-ctype             v1.25.0    Symfony polyfill for ctype functions
symfony/polyfill-iconv             v1.25.0    Symfony polyfill for the Iconv extension
symfony/polyfill-intl-grapheme     v1.25.0    Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-idn          v1.25.0    Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer   v1.25.0    Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring          v1.25.0    Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.25.0    Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73             v1.25.0    Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/polyfill-php80             v1.25.0    Symfony polyfill backporting some PHP 8.0+ features to lower PHP versions
symfony/polyfill-php81             v1.25.0    Symfony polyfill backporting some PHP 8.1+ features to lower PHP versions
symfony/process                    v5.4.5     Executes commands in sub-processes
symfony/psr-http-message-bridge    v2.1.2     PSR HTTP message bridge
symfony/routing                    v5.4.3     Maps an HTTP request to a set of configuration variables
symfony/service-contracts          v2.5.0     Generic abstractions related to writing services
symfony/string                     v5.4.3     Provides an object-oriented API to strings and deals with bytes, UTF-8 code points and grapheme clusters in a unified way
symfony/translation                v5.4.6     Provides tools to internationalize your application
symfony/translation-contracts      v2.5.0     Generic abstractions related to translation
symfony/var-dumper                 v5.4.6     Provides mechanisms for walking through any arbitrary PHP variable
tecnickcom/tcpdf                   6.4.4      TCPDF is a PHP class for generating PDF documents and barcodes.
theseer/tokenizer                  1.2.1      A small library for converting tokenized PHP source code into XML and potentially other formats
tijsverkoyen/css-to-inline-styles  2.2.4      CssToInlineStyles is a class that enables you to convert HTML-pages/files into HTML-pages/files with inline styles. This is very useful when you're sending emails.
vlucas/phpdotenv                   v5.4.1     Loads environment variables from `.env` to `getenv()`, `$_ENV` and `$_SERVER` automagically.
voku/portable-ascii                1.6.1      Portable ASCII library - performance optimized (ascii) string functions for php.
webmozart/assert                   1.10.0     Assertions to validate method input/output with nice error messages.
```

**Comment by aimeos** [comment]
Look OK. Do you have a stack trace of the problem?

**Comment by cutemonster** [comment]
do you mean the Exception trace from cmd prompt?

```
Exception trace:
  at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/process/Process.php:143
 Symfony\Component\Process\Process->__construct() at C:\111Projects\WEB\cueforce\app\Composer.php:210
 App\Composer::executeCommand() at C:\111Projects\WEB\cueforce\app\Composer.php:178
 App\Composer::setup() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:391
 Composer\EventDispatcher\EventDispatcher->executeEventPhpScript() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:248
 Composer\EventDispatcher\EventDispatcher->doDispatch() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:125
 Composer\EventDispatcher\EventDispatcher->dispatchScript() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Command/CreateProjectCommand.php:321
 Composer\Command\CreateProjectCommand->installProject() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Command/CreateProjectCommand.php:167
 Composer\Command\CreateProjectCommand->execute() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:1015
 Symfony\Component\Console\Application->doRunCommand() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:299
 Symfony\Component\Console\Application->doRun() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:334
 Composer\Console\Application->doRun() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:130
 Composer\Console\Application->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/bin/composer:83
 require() at C:\ProgramData\ComposerSetup\bin\composer.phar:29

create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--add-repository] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--remove-vcs] [--no-install] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--ask] [--] [<package> [<directory> [<version>]]]
```

**Comment by aimeos** [comment]
The problem is caused by composer 2.3.x, which contains a newer Symfony Process implementation which is incompatible to the one used before. We've fixed that and created a new release (2021.10.5) which installs without problems now when using composer 2.3

**Comment by cutemonster** [comment]
Thanks!
